src/femtomes.h

```diff
@@ -0,0 +1,534 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2012-2018 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <math.h>
+#include <string.h>
+#include <ctime>
+#include <cmath>
+
+#include "femtomes.h"
+#include "rtcm.h"
+
+#ifndef M_PI_F
+# define M_PI_F 3.14159265358979323846f
+#endif
+
+#define MIN(X,Y)	((X) < (Y) ? (X) : (Y))
+#define FEMTO_UNUSED(x) (void)x;
+
+#if defined _FMTOMES_DEBUG
+#define _FEMTOMES_DEBUG_LEVEL	2
+#else
+#define _FEMTOMES_DEBUG_LEVEL	2
+#endif
+
+#if _FEMTOMES_DEBUG_LEVEL
+	#define FEMTO_INFO(...)			{GPS_INFO(__VA_ARGS__);}
+	#define FEMTO_DEBUG(...)		{GPS_WARN(__VA_ARGS__);}
+	#define FEMTO_ERR(...)			{GPS_ERR(__VA_ARGS__);}
+#else
+	#define FEMTO_INFO(...)			{(void)0;}
+	#define FEMTO_DEBUG(...)		{(void)0;}
+	#define FEMTO_ERR(...)			{(void)0;}
+#endif
+
+
+GPSDriverFemto::GPSDriverFemto(GPSCallbackPtr callback, void *callback_user,

```

Comment:
Why is this code in the header file? It is duplicated from the .cpp file.
Also the class declaration for `GPSDriverFemto` is missing, so this will not compile, and the pull request is incomplete in this state.

Reply:
so sorry i can't understand about your answer.
1、 the 68 line will complete the init work when there have GPSDriverFemto object be create
2、this class declaration have done in the femtomes.h, you see the 68 line is in femtomes.cpp
3、I have do test about this code, we have log and video. so it sure to can be compiled.

Reply:
Have a look at femtones.h and femtones.cpp in this PR: they both contain the same content, so it cannot work.

Reply:
I'm sorry about have not notice the commit file of femtomes.h same to the femtomes.cpp. Later we will fix this problem. Thank you.

Reply:
I have recommit the femtomes.h, please check this again,thank you.